### ISSUE TITLE
Issue #1309 Can't resolve './app.module.ngfactory' is still present for many of us

### ISSUE THREAD

**Issue by mburger81** [ionic-team/ionic-app-scripts#1315]
#### Short description of the problem:
As described several days ago in post #1309 the bug is still present for many of us, we did many posts there also I created a repo to reproduce the scenario. Perhaps it is better I create a new one because the other one is closed, so this is what I have wrote there:

On my linux machine It's really simple to simulate the bug which for me still persists.
I created an empty ionic project with `ionic start` selecting the `blank` starter template.

After that I remove the `Home` component from `app.module.ts`, this component is not loaded and the build script should give error

I created a repo to help you:
https://github.com/mburger81/ionic-ng5-bug

On running`./node_modules/.bin/ngc` I get this error
`Error: Cannot determine the module for class HomePage in ~/ionic-ng5-bug/src/pages/home/home.ts! Add HomePage to the NgModule to fix it.`

On running ionic `cordova build android --prod --release` I get always the same error
```
Error: ./src/app/main.ts
Module not found: Error: Can't resolve './app.module.ngfactory' in '~/ionic-ng5-bug/src/app'
resolve './app.module.ngfactory' in '~/ionic-ng5-bug/src/app'
  using description file: ~/ionic-ng5-bug/package.json (relative path: ./src/app)
    Field 'browser' doesn't contain a valid alias configuration
  after using description file: ~/ionic-ng5-bug/package.json (relative path: ./src/app)
    using description file: ~/ionic-ng5-bug/package.json (relative path: ./src/app/app.module.ngfactory)
      no extension
        Field 'browser' doesn't contain a valid alias configuration
        ~/ionic-ng5-bug/src/app/app.module.ngfactory doesn't exist
      .ts
        Field 'browser' doesn't contain a valid alias configuration
        ~/ionic-ng5-bug/src/app/app.module.ngfactory.ts doesn't exist
      .js
        Field 'browser' doesn't contain a valid alias configuration
        ~/ionic-ng5-bug/src/app/app.module.ngfactory.js doesn't exist
      .json
        Field 'browser' doesn't contain a valid alias configuration
        ~/ionic-ng5-bug/src/app/app.module.ngfactory.json doesn't exist
      as directory
        ~/ionic-ng5-bug/src/app/app.module.ngfactory doesn't exist
[~/ionic-ng5-bug/src/app/app.module.ngfactory]
[~/ionic-ng5-bug/src/app/app.module.ngfactory.ts]
[~/ionic-ng5-bug/src/app/app.module.ngfactory.js]
[~/ionic-ng5-bug/src/app/app.module.ngfactory.json]
[~/ionic-ng5-bug/src/app/app.module.ngfactory]
 @ ./src/app/main.ts 2:0-60
    at BuildError.Error (native)
    at new BuildError (~/ionic-ng5-bug/node_modules/@ionic/app-scripts/dist/util/errors.js:16:28)
    at callback (~/ionic-ng5-bug/node_modules/@ionic/app-scripts/dist/webpack.js:121:28)
    at emitRecords.err (~/ionic-ng5-bug/node_modules/webpack/lib/Compiler.js:269:13)
    at Compiler.emitRecords (~/ionic-ng5-bug/node_modules/webpack/lib/Compiler.js:375:38)
    at emitAssets.err (~/ionic-ng5-bug/node_modules/webpack/lib/Compiler.js:262:10)
    at applyPluginsAsyncSeries1.err (~/ionic-ng5-bug/node_modules/webpack/lib/Compiler.js:368:12)
    at next (~/ionic-ng5-bug/node_modules/tapable/lib/Tapable.js:218:11)
    at Compiler.compiler.plugin (~/ionic-ng5-bug/node_modules/webpack/lib/performance/SizeLimitsPlugin.js:99:4)
    at Compiler.applyPluginsAsyncSeries1 (~/ionic-ng5-bug/node_modules/tapable/lib/Tapable.js:222:13)
```

**This is my environment:**
```
cli packages: (/usr/lib/node_modules)
    @ionic/cli-utils  : 1.18.0
    ionic (Ionic CLI) : 3.18.0

global packages:
    cordova (Cordova CLI) : 7.1.0 

local packages:
    @ionic/app-scripts : 3.1.2
    Cordova Platforms  : android 6.3.0
    Ionic Framework    : ionic-angular 3.9.2

System:
    Android SDK Tools : 26.1.1
    Node              : v6.11.5
    npm               : 5.5.1 
    OS                : Linux 4.13

Environment Variables:
    ANDROID_HOME : ~/android/sdk

Misc:
    backend : legacy

```

**Edit:**
I updated also to latest `Node LTS version 8.9.1` with the same result

**Comment by chrisvanderkooi** [comment]
```
Error: Cannot determine the module for class HomePage in ~/ionic-ng5-bug/src/pages/home/home.ts! Add HomePage to the NgModule to fix it.
```

This error literally tells you the issue. You need to add the HomePage class to the src/app/app.module.ts in the declarations and entryComponents arrays.

In ng5, all components (@Component) needs to be added to a module in order to build.

**Comment by mburger81** [comment]
@chrisvanderkooi thx but are you kidding me?
first: I don't need help to create the app.module.ts!!! I have done this example to reproduce the  issue!!! So it is my intention there is no HomePage component loaded on no module!!
second: the error you have seen  in my post  is  NOT the error from app scripts  / ionic but from NGC!!!!!!

So sorry, thx for your intention to help, but your post is completly wrong! Thx  👍 

Bug is still there and meantime there is another open bug! #1317

**Comment by flakolefluk** [comment]
@lucastelnovo Take a look at this
https://github.com/ionic-team/starters/pull/20
it has been merged now. i guess it could be your  issue with  "super" template

**Comment by danbucholtz** [comment]
This is fixed in `3.1.4`.

Thanks,
Dan
